### PR TITLE
addition creating candidate

### PR DESCRIPTION
**Pull request by Fevor7**

### CODE REVIEW

File: frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.css [hunk 1]
```diff
@@ -0,0 +1,40 @@
+.candidateForm {
```
Comment (yauhen-papko) [inline frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.css]: do not use camelCase in css selectors
Few links for reference:
http://bdavidxyz.com/blog/how-to-name-css-classes/
http://bdavidxyz.com/blog/how-to-name-css-classes/

File: frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+@Component({
+  selector: 'app-candidate-form',
+  templateUrl: './candidate-form.component.html',
+  styleUrls: ['./candidate-form.component.css']
+})
+export class CandidateFormComponent implements OnInit, OnDestroy {
+
+  @Output() displayForm = new EventEmitter();
+  public candidate: Candidate = new Candidate();
```
Comment (yauhen-papko) [inline frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.ts]: 'public' is not required.

File: frontend/src/app/interceptor/authentication-interceptor.ts [hunk 3]
```diff
@@ -23,30 +25,29 @@ export class AuthenticationInterceptor implements HttpInterceptor {
...
+        });
+        } else {
           this.auth.removeCredentialsUser();
           this.redirectToSignIn("don't have refreshToken");
         }
       }
-    }else{
-      if(req.url === 'http://localhost:8080/interview-portal/auth'){
+    } else {
+      if (req.url === 'http://localhost:8080/interview-portal/auth') {
```
Comment (yauhen-papko) [inline frontend/src/app/interceptor/authentication-interceptor.ts]: Will this code work in non-localhost environment?

Comment (Fevor7) [reply frontend/src/app/interceptor/authentication-interceptor.ts]: No, I will enter proxy. I don't have time in the sprint.

File: frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.ts (hunk 2 above)
Comment (Fevor7) [reply frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.ts]: Yes, he is not required. I wanted use him in template, but now I using reactive forms. I will correct it.

File: frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.css (hunk 1 above)
Comment (Fevor7) [reply frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.css]: Thank you for information. I will correct.

File: frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.ts (hunk 2 above)
Comment (yauhen-papko) [reply frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.ts]: he -> it;

File: frontend/src/app/api/models/work-candidate.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+export class WorkCandidate {
+
+  dateEnd?: string;
+
+  dateStart?: string;
+
+  id?: number;
+
+  nameCompany?: string;
```
Comment (aliaksandr-aliaksandrau) [inline frontend/src/app/api/models/work-candidate.ts]: Please rename these fields to 'companyName', 'endDate', 'startDate' and so on.
Read more about 'noun adjunct': 
https://en.wikipedia.org/wiki/Noun_adjunct
https://glossary.sil.org/term/noun-adjunct

File: frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.css [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+sup {
+    color: rgb(255, 0, 0);
+}
+
+select {
+    float: left;
+    width: 92%;
+}
```
Comment (aliaksandr-aliaksandrau) [inline frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.css]: Please add emty line at the end of the file.

File: frontend/src/app/interceptor/authentication-interceptor.ts [hunk 6]
```diff
@@ -23,30 +25,29 @@ export class AuthenticationInterceptor implements HttpInterceptor {
...
+      else {
         this.redirectToSignIn("don't have Token");
       }
-     
     }
   };
   private redirectToSignIn(error) {
     this.router.navigate(['/login']);
-    return Observable.throw(error); //** Нужно пофиксить обработку */
+    return Observable.throw(error); //* Нужно пофиксить обработку /
```
Comment (aliaksandr-aliaksandrau) [inline frontend/src/app/interceptor/authentication-interceptor.ts]: Please add comments only in English.

File: frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.ts [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+  }
+
+  fetchDisciplines(): any {
+    this.disciplinesService.findAllUsingGET()
+      .takeUntil(this.destroy)
+      .subscribe((disciplines) => {
+        this.disciplines = disciplines;
+      }, (error) => {
+        console.log('Send to error page when it appears');
+      });
```
Comment (aliaksandr-aliaksandrau) [inline frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.ts]: In such cases the arrow function can have the following syntax:
error => console.log('....')

File: frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.html [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+        <div class="text-center add-candidates">
+            <a class="btn btn-primary" (click)="createCandidate()">Add Candidate</a>
+            <a class="btn btn-secondary" (click)="change()">Close</a>
+            <br>
+            <br>
+            <span *ngIf="displayErrorMessage" class="error-message">Fill in all the fields</span>
+        </div>
+    </form>
+</div>
+<app-message-window *ngIf="displayMessage" [message]="viewMessage" (displayWindowMessage)="change()"></app-message-window>
```
Comment (aliaksandr-aliaksandrau) [inline frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.html]: This template is too big and there is a lot of repeating code. Maybe it makes sense to create a special component.

File: frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CandidateFormComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
```
Comment (aliaksandr-aliaksandrau) [inline frontend/src/app/candidates/candidate-list/candidate-form/candidate-form.component.spec.ts:25]: Please add tests for logic, not only for creation.

**Pull request merged**